src/dbus_api/api.rs

```diff
@@ -56,27 +57,31 @@ fn create_pool(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 
     let object_path = m.path.get_name();
     let dbus_context = m.tree.get_data();
-    let result = dbus_context
-        .engine
-        .borrow_mut()
-        .create_pool(name, &blockdevs, tuple_to_option(redundancy), force);
+    let mut engine = dbus_context.engine.borrow_mut();
+    let result = engine.create_pool(name, &blockdevs, tuple_to_option(redundancy), force);
 
     let return_message = message.method_return();
 
     let msg = match result {
-        Ok((uuid, devnodes)) => {
+        Ok(pool_uuid) => {
             let pool_object_path: dbus::Path =
-                create_dbus_pool(dbus_context, object_path.clone(), uuid);
-            let paths = devnodes
-                .iter()
-                .map(|d| {
-                         d.to_str()
-                             .expect("'d' originated in the 'devs' D-Bus argument")
-                             .into()
-                     });
-            let paths = paths.map(MessageItem::Str).collect();
+                create_dbus_pool(dbus_context, object_path.clone(), pool_uuid);
+            let return_sig = "(oao)";
+            let default_return = MessageItem::Array(vec![], return_sig.into());
```

Comment:
I don't think this is correct default return. It's an array with items with type ```(oao)```. It should be more like:
```MessageItem::Struct(vec![MessageItem::Array(vec![], ...```.

Reply:
ok I think I got this right, please take a look.